app/battle/battle.py

```diff
@@ -0,0 +1,31 @@
+class Battle:
+
+    def __init__(self, knights_dict: dict):
+        self.knights_dict = knights_dict
+
+    def versus(self, knight_1: str, knight_2: str):
+        result_dict = {}
+        if self.knights_dict.get(knight_1) and self.knights_dict.get(knight_2):
+            self.knights_dict[knight_1].name['hp'] -= \
+                self.knights_dict[knight_2].name['power'] \
+                - self.knights_dict[knight_1].name['protection']
+            self.knights_dict[knight_2].name['hp'] -= \
+                self.knights_dict[knight_1].name['power'] \
+                - self.knights_dict[knight_2].name['protection']
```

Comment:
DRY: Don't Repeat Yourself
WET: Write Everything Twice

I believe - you choose WET here, instead of DRY?

Reply:
where DRY? I’m counting for first and second HP

Reply:
Hm, okay, it can be a method of Knight - to beat different Knight, but I agree with - this implementation is also ok